### PR TITLE
UI: Add DAS summary table filtering

### PR DESCRIPTION
**Pull request by backspace**
This adds text and facet filtering to the DAS summary table. It’s mostly a combination
of facet code taken from `controller:jobs/index`, search adapted from the global
search control, and tests.

The only search is fuzzy matching against the recommendation summary slug; the other
strings of task names and namespaces didn’t seem applicable to me, but it wouldn’t
be difficult to add anything else desirable.

Some design decisions:
* if changing the filter causes the currently-active summary to be excluded, the active card (and subroute) updates to the first summary in the filtered collection
* if the filter causes the collection to become empty, it navigates to `/optimize?…`

### CODE REVIEW

File: ui/tests/acceptance/optimize-test.js [hunk 1]
```diff
@@ -351,6 +351,339 @@ module('Acceptance | optimize', function(hooks) {
...
+          groupTaskCount: 2,
+          childrenCount: 0,
+        });
+      });
+      await Optimize.visit();
+    },
+    filter: (taskGroup, selection) => selection.find(prefix => taskGroup.job.name.startsWith(prefix)),
+  });
+
+  function testFacet(label, { facet, paramName, beforeEach, filter, expectedOptions }) {
```
Comment (DingoEatingFuzz) [inline ui/tests/acceptance/optimize-test.js:604]: This is another one of those things where there just wasn't quite a good way to abstract it when I introduced the pattern for clients and jobs. Would be nice to simplify somehow.

**Review by DingoEatingFuzz — APPROVED** [review]
Nice work on this. I think the facet code is a bit complex to have duplicated, but that doesn't have to block this.

I did find one minor bug: when filtering from a selection of no results back to one with results, the active recommendation gets no card.

**To reproduce this with the preview on this PR...**
1. Filter by Status > Dead
2. See no results
3. Remove Status filter
4. See results with the first recommendation in the active state but no card

Since the issue is minor, it's totally okay to fix it after merging this.

**Comment by DingoEatingFuzz** [comment]
Merging this to get it into the next beta. 🇨🇦

File: ui/app/controllers/optimize.js [hunk 2]
```diff
@@ -1,20 +1,165 @@
...
+      // eslint-disable-next-line ember/no-side-effects
+      this.qpPrefix = serialize(intersection(availablePrefixes, this.selectionPrefix));
+    });
+
+    // Sort, format, and include the count in the label
+    return prefixes.sortBy('prefix').map(name => ({
+      key: name.prefix,
+      label: `${name.prefix} (${name.count})`,
+    }));
+  }
```
[the remark this answers is not in the document]
Comment (backspace) [reply ui/app/controllers/optimize.js:118]: sure, sounds good to me!

**Comment by backspace** [comment]
Thanks for the review, I’ve added todos for the facets duplication and bug that you found.

**Pull request merged**